### DISCUSSION TITLE
Layout commands

### DISCUSSION THREAD

**Discussion by mahume** [zellij-org/zellij#2841] [Q&A] [answered]
In a layout I'm seeing the ability to send commands. I'm enjoying the functionality of these commands that can be suspended, exited or rerun simply by hitting enter. I can see this being very useful. But how about sending just a basic command into a pane on startup and letting it exit to display the prompt again? For example, let's say I wanted a tab that would open a pane, switch my node version, then open nvim automatically or manually. 

```
...
    tab name="Project" {
        pane {
            cwd "/Users/username/repos/my-repo"
            command "fnm"
            args "use" "18"
        }
    }
...
```

This tab essentially becomes a way to run fast node manager over and over again by pressing enter. Instead of sending the command, then providing the prompt to run another command. Such as nvim. 

Is the ability to send one off commands possible as of now?

**Comment by Lockszmith-GH** [comment]
Do you mean [close_on_exit](https://zellij.dev/documentation/creating-a-layout.html?highlight=command#close_on_exit)? If so, it's clearly documented.

**Reply by mahume** [reply]
I did see that but it's not applicable. That closes the pane when its command exits. Which would be good if you only want to send a one shot command and then close that pane. But sometimes you need the ability to run a command before another one. Such as switching your Node version before running another command. Or just keep the pane open for other tasks.

Tmux bash script example which sends a command to switch Node versions when starting up. The pane is still there to do whatever you please.

```
# Send keys to window's top pane (first pane of window)
tmux select-pane -t $SESSION:Window.1
tmux send-keys -t $SESSION:Window.1 'fnm use 18' C-m
```

**Comment by imsnif** [comment, marked as answer]
How about offloading this to the shell? Something like:

```
    tab name="Project" {
        pane {
            cwd "/Users/username/repos/my-repo"
            command "bash"
            args "-c" "fnm use 18; bash"
        }
    }
```

I don't want to implement something that sends keys to panes because I think it's very hacky and not reproducible across systems. I do plan on adding something like `<ESC> - drop to shell` as a third key shortcut to command panes, if that will do the trick for you?

**Reply by mahume** [reply]
That solution works. And I agree, that would be a very useful feature. Especially within a custom layout where you just need to pop out for a second and do something in that directory before hopping back into that command pane.

Also great work on Zellij. I'm following it closely. Once the new Sessions feature gets more nailed down I'm gonna make the Tmux -> Zellij leap.
